File: strings/reverse_prefix_of_word.cs
public class Solution {
    public string ReversePrefix(string word, char ch) {
        if (word.Length == 1) return word;

        int idx = word.IndexOf(ch);
        if (idx == -1) return word;

        StringBuilder sb = new(word.Length);
        for (int i = idx; i >= 0; --i) sb.Append(word[i]);
        for (int i = idx + 1; i < word.Length; ++i) sb.Append(word[i]);

        return sb.ToString();
    }
}

File: strings/reverse_prefix_of_word.cpp
class Solution {
public:
    string reversePrefix(string word, char ch) {
        if (word.size() == 1) return word;

        int idx = word.find_first_of(ch);
        if (idx == -1) return word;

        string result = {};
        for (int i = idx; i >=0; --i) result += word[i];
        for (int i = idx + 1; i < word.size(); ++i) result +=word[i];

        return result;
    }
};
